### DISCUSSION TITLE
Is this expected behaviour?

### DISCUSSION THREAD

**Discussion by lightiverson** [Snaipe/Criterion#460] [Q&A] [answered]
It seems when passing a user defined exit code to a test, Criterion ignores any other assertions. Take the attached screenshot for example, shouldn't this test fail because of the false assertion?
<img width="1493" alt="Schermafbeelding 2022-09-06 om 18 36 51" src="https://user-images.githubusercontent.com/33266555/188689478-c99e2933-be2a-4a2f-82ff-37657768ba32.png">

**Comment by Snaipe** [comment, marked as answer]
Your `cr_assert_stderr_eq_str` never runs, because error() exits the process by calling exit(0). This behavior is quite normal.

**Reply by lightiverson** [reply]
Ah, that makes sense. Thanks!

**Comment by lightiverson** [comment]
Is there any way to check for `cr_assert_stderr_eq_str` as well as a custom `.exit_code` coming from the same function?

**Reply by Snaipe** [reply]
Well, if cr_assert_eq_str runs before your test exits, then yes, it would work. .exit_code is simply a way to check that the entire test must exit with a specific exit status, but you can run any number of assertions between the start and the call to exit.
